doc/developer-guides/security-hld.rst

```diff
@@ -0,0 +1,1081 @@
+
+Normally, this warm reboot (crashdump) feature is a debug feature, and
+must be disabled in a production release. User who wants to use this
+feature must possess the private signing key to re-sign the image (e.g.
+the virtual SBL image) after enabling the configuration.
+
+.. _uos_suspend_resume:
+
+UOS Suspend/Resume
+~~~~~~~~~~~~~~~~~~
+
+There is no special design considerations for normal UOS without secure
+world supported, as long as the EPT/VT-d memory protection/isolation is
+active during the entire suspended time.
+
+Secure world (Trusty/TEE) is a special case for virtual suspend. Unlike
+the non-secure world of UOS, whose memory content can be read/written by
+SOS, the memory content of secure world of UOS must not be visible to
+SOS. This is designed for security with defense in depth.
+
+During the entire process of UOS sleep/suspend, the memory protection
+for secure-world must be preserved too.The physical memory region of
+secure world must be removed from EPT paging tables of any guest VM,
+even including the SOS VM.
+
+Third-party libraries
+---------------------
+
+All the third-party libraries must be examined before use to verify
+there are no known vulnerabilities in the library source code.
+Typically, the CVE site https://cve.mitre.org/cve/search_cve_list.html
+can be used to search known vulnerabilities.
+
+.. _platform_root_of_trust:
+
+Platform Root of Trust Key/SEED Derivation
+==========================================
+
+For security reason, each guest VM requires a root key, is used to
+derive many other individual keys for different purposes, for example,
+secure storage encryption, keystore master key, and HMAC keys.
+
+In the APL platform, CSE FW will generate platform SEED (pSEED, 256bit)
+unique per device since it is derived from a unique chipset secret
+burned into the chip.
+
+Then on each boot, the SBL BIOS is responsible for retrieving the pSEED
+from CSE FW, and deriving two other derivatives (dSEED, and uSEED).
+
+.. figure:: images/security-image6.png
+   :width: 900px
+   :align: center
+   :name: security-seed
+
+   Platform SEED (pSEED) Derivation
+
+As shown in :numref:`security-seed` above, the hypervisor then derives
+multiple child SEEDs for multiple guest VMs. A guest VM must not be able
+to know the SEEDs of any other guest VMs.
+
+The algorithm used in the hypervisor to derive keys is HKDF (HMAC-based
+Extract-and-Expand Key Derivation Function, `RFC5869
+<https://tools.ietf.org/html/rfc5869>`_.  The crypto library `mbedtls
+<https://github.com/ARMmbed/mbedtls>`_ is chosen for project ACRN.
+
+The parameters of HDKF derivation in the hypervisor are:
+
+#. VMInfo= vm-uuid (from hypervisor configuration file)
+#. theHash=SHA-256
+#. OutSeedLen = 64 in bytes
+#. Guest Dev and User SEED (dvSEED/uvSEED)
+
+   dvSEED = HKDF(theHash, nil, dSEEd, VMInfo\|"devseed", OutSeedLen)
+
+   uvSEED = HKDF(theHash, nil, uSEEd, VMInfo\|"userseed", OutSeedLen
+
+.. _secure_trusty:
```

Comment:
Should this also be a good section to include a pointer to https://github.com/projectacrn/acrn-hypervisor/blob/master/doc/developer-guides/trusty.rst?

@bzhu5

Reply:
but this is done internally in hypervisor when hcall_initialize_trusty() is called.

Reply:
I do not understand your previous comment?

I was suggesting that we add a link to this other document we have that talks about Trusty.